src/main/java/io/supertokens/webserver/api/emailverification/UnverifyEmailAPI.java

```diff
@@ -52,13 +58,22 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         email = Utils.normaliseEmail(email);
 
         try {
-            EmailVerification.unverifyEmail(this.getAppIdentifierWithStorage(req), userId, email);
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage;
+            try {
+                StorageAndUserIdMapping storageAndUidMapping = getStorageAndUserIdMappingForAppSpecificApi(
+                        req, userId, UserIdType.ANY);
+                storage = storageAndUidMapping.storage;
+            } catch (UnknownUserIdException e) {
+                storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+            }
```

Comment:
Actually, do we need to care about user id mapping here? Cause this is a non auth recipe. We need to be careful about when we use user id mapping and when we don't. In general, we only use it in auth recipe and stuff like delete user, make user primary. But i think here is needed cause we mark users as verified in the sign up apis based on the supertokens user id right? But earlier it seemed like we used to not resolve user id mapping here, so how did things work, and this change may break stuff for users now (since they may have external user id & email marked as verified as opposed to supertokens user id)

Reply:
I think this was a bug in the master branch to. Try this out: Create a passwordless user in the core -> call the user id mapping api -> call the isEmailVerified with the supertokens user id (should return true), but then call with the external user id (should return false). Lmk if what i am thinking is correct or not.